src/compiler/scala/tools/nsc/typechecker/TypeDiagnostics.scala

```diff
@@ -518,11 +518,19 @@ trait TypeDiagnostics {
         (sym ne null)
           && (sym.isTerm && qualifiesTerm(sym) || sym.isType && qualifiesType(sym))
         )
-
-      override def traverse(t: Tree): Unit = if (!t.isErrorTyped) {
+      def isExisting(sym: Symbol) = sym != null && sym.exists
+
+      private var sawErrorTyped: Boolean = false
+      def errorAware(t: Tree)(body: => Unit): Unit = {
+        val saveErrorTyped = sawErrorTyped
+        sawErrorTyped = t.isErrorTyped
+        body
+        sawErrorTyped = saveErrorTyped
+      }
+      override def traverse(t: Tree): Unit = errorAware(t) {
```

Comment:
Instead of stacking up the `saveErrorTyped`-s, wouldn't it be equivalent to just define here
```
def sawErrorTyped = t.isErrorTyped
```
? (And then inline that).

Reply:
I was being paranoid about the state below an error, but I'll do more experimentation. (I was also trying to avoid experimentation with this moribund code.)